src/main/java/io/logz/guice/jersey/JerseyServer.java

```diff
@@ -76,4 +99,26 @@ protected Injector getInjector() {
         server.setHandler(webAppContext);
     }
 
+    private void validate(JettyConfiguration jettyConfiguration) {
+        if (jettyConfiguration.getMinThreads() < 1) {
+            throw new ConfigurationException(String.format("minThreads (%s) must be bigger than 0", jettyConfiguration.getMinThreads()));
+        }
+        if (jettyConfiguration.getMaxThreads() < jettyConfiguration.getMinThreads()) {
+            throw new ConfigurationException(String.format("maxThreads (%s) must be bigger than minThreads (%s)", jettyConfiguration.getMaxThreads(), jettyConfiguration.getMinThreads()));
+        }
+        if (jettyConfiguration.getIdleTimeout() < 0) {
+            throw new ConfigurationException(String.format("idleTimeout (%s) must be 0 or bigger", jettyConfiguration.getIdleTimeout()));
+        }
+        if (jettyConfiguration.getMaxQueueSize() > jettyConfiguration.getMaxThreads()) {
+            LOGGER.warn("maxQueueSize ({}) is bigger than the maxThreads ({}) - this may lead to server hanging onto queued requests (potentially long after the client has timed out)",
+                    jettyConfiguration.getMaxQueueSize(), jettyConfiguration.getMaxThreads());
+        }
+    }
+
+    private class ConfigurationException extends RuntimeException {
```

Comment:
Not nice throwing special exception but not allowing the user to add catch block for this type of exceptions 😉